.github/workflows/rpcs3.yml

```diff
@@ -104,6 +104,91 @@ jobs:
           export AVVER="${COMM_TAG}-${COMM_COUNT}"
           .ci/github-upload.sh
 
+  Mac_Build:
+    # Only run push event on master branch of main repo, but run all PRs
+    if: github.event_name != 'push' || (github.repository == 'RPCS3/rpcs3' && github.ref_name == 'master')
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - name: Intel
+            build_sh: "arch -X86_64 .ci/build-mac.sh"
+            UPLOAD_COMMIT_HASH: 8e21bdbc40711a3fccd18fbf17b742348b0f4281
+            UPLOAD_REPO_FULL_NAME: "rpcs3/rpcs3-binaries-mac"
+          - name: Apple Silicon
+            build_sh: ".ci/build-mac-arm64.sh"
+            UPLOAD_COMMIT_HASH: 51ae32f468089a8169aaf1567de355ff4a3e0842
+            UPLOAD_REPO_FULL_NAME: "rpcs3/rpcs3-binaries-mac-arm64"
+    name: RPCS3 Mac ${{ matrix.name }} 
+    runs-on: macos-14
+    env:
+      CCACHE_DIR: "/tmp/ccache_dir"
+      CI_HAS_ARTIFACTS: true
+      QT_VER: '6.7.3'
+      QT_VER_MAIN: '6'
+      LLVM_COMPILER_VER: '19'
+      RELEASE_MESSAGE: "../GitHubReleaseMessage.txt"
+      UPLOAD_COMMIT_HASH: ${{ matrix.UPLOAD_COMMIT_HASH }}
+      UPLOAD_REPO_FULL_NAME: ${{ matrix.UPLOAD_REPO_FULL_NAME }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@main
+        with:
+          fetch-depth: 0
+
+      - name: Setup Cache
+        uses: actions/cache@main
+        with:
+          path: ${{ env.CCACHE_DIR }}
+          key: ${{ runner.os }}-ccache-${{ matrix.name }}-${{github.run_id}}
+          restore-keys: | 
+            ${{ runner.os }}-ccache-${{ matrix.name }}-
+
+      - name: Setup Homebrew Cache
+        uses: actions/cache@main
+        with:
+          path: "~/Library/Caches/Homebrew"
+          key: ${{ runner.os }}-homebrew-${{ matrix.name }}
```

Comment:
So this cache command will only ever save one cache, the very first one, slowly getting outdated to the point where it's progressively useless.
We can try storing progressive caches, but our cache storage is very limited and we already have big caches on other pipelines. So my question is:
- How big does a homebrew cache get?
- How much CI time are we saving by caching this?

Reply:
I checked the cache size of macOS-homebrew-Intel, looks like it clocks in at 730 MB, pretty hefty. and another 580 MB for Apple Silicon. So I guess the question is, is the time savings we get from the ~1.5 GB cache size (we have a max of 10 GBs across all pipelines) worth it?